pkg/controller/daemon/daemon_controller.go

```diff
@@ -777,6 +802,7 @@ func (dsc *DaemonSetsController) podsShouldBeOnNode(
 	nodeToDaemonPods map[string][]*v1.Pod,
 	ds *apps.DaemonSet,
 	hash string,
+	logger klog.Logger,
```

Comment:
IIRC I have mostly seen the logger to be the first argument of the function - might be better to do it here as well